sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/TagEnumerationState.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using OpenTelemetry.Trace;
+
+namespace Microsoft.OpenTelemetry.Exporter.AzureMonitor
+{
+    internal struct TagEnumerationState : IActivityEnumerator<KeyValuePair<string, object>>
```

Comment:
Just for my understanding, what is an `IActivityEnumerator` ?

Reply:
It's an interface in OpenTelemetry to perform zero allocation. Ref: https://github.com/open-telemetry/opentelemetry-dotnet/blob/dc3752f55d3b33dc277c0b67c842fba06a89114b/src/OpenTelemetry.Api/Internal/IActivityEnumerator.cs#L25

Reply:
Just to clarify: These has nothing to do with OpenTelemetry, except that we used it in OpenTelemetry as well.
There are some tricks to avoid IEnumerable allocation.